src/aggregator/aggregator/follower_flush_mgr.go

```diff
@@ -346,18 +346,18 @@ func (mgr *followerFlushManager) standardFlushersFromKVUpdateWithLock(
 		shardFlushTimes, exists := mgr.received.ByShard[shard]
 		if !exists {
 			metrics.shardNotFound.Inc(1)
-			logger.WithFields(
-				log.NewField("shard", shard),
+			logger.With(
+				zap.Uint32("shard", shard),
 			).Warn("shard not found in flush times")
 			continue
 		}
 		flushTimes := getFlushTimesByResolutionFn(shardFlushTimes)
 		lastFlushedAtNanos, exists := flushTimes[int64(resolution)]
 		if !exists {
 			metrics.resolutionNotFound.Inc(1)
-			logger.WithFields(
-				log.NewField("shard", shard),
-				log.NewField("resolution", resolution.String()),
+			logger.With(
+				zap.Uint32("shard", shard),
+				zap.Stringer("resolution", resolution),
```

Comment:
neat

Reply:
Yeah! Zap's perf numbers are pretty great because of these API choices. https://github.com/uber-go/zap#performance